src/PhpDoc/TypeNodeResolver.php

```diff
@@ -904,6 +908,14 @@ private function resolveConstTypeNode(ConstTypeNode $typeNode, NameScope $nameSc
 		return new ErrorType();
 	}
 
+	private function resolveOffsetAccessNode(OffsetAccessTypeNode $typeNode, NameScope $nameScope): Type
+	{
+		$type = $this->resolve($typeNode->type, $nameScope);
+		$offset = $this->resolve($typeNode->offset, $nameScope);
+
+		return new OffsetAccessType($type, $offset);
```

Comment:
TypeNodeResolver could detect when the type doesn't make sense. For example when `$type` isn't offset-accessible or when `$type–>hasOffsetValueType($offset)` is `no()`. It could return `new ErrorType()` in that case and the appropriate rules will pick this ErrorType and complain about an "unresolvable type" automatically.